packages/core/src/services/commits/runDocumentAtCommit.ts

```diff
@@ -179,10 +180,29 @@ async function iterate({
         logHandler,
       })
     }
-  } catch (error) {
-    controller.error(error)
-
-    throw error
+  } catch (e) {
+    const error = e as Error
+    enqueueEvent(controller, {
+      event: StreamEventTypes.Latitude,
+      data: {
+        type: ChainEventTypes.Error,
+        error: {
+          name: error.name,
+          message: error.message,
+          stack: error.stack,
+        },
+      },
+    })
+    controller.close()
+    return {
+      text: error.message,
+      usage: {
+        promptTokens: 0,
+        completionTokens: 0,
+        totalTokens: 0,
+      },
+      toolCalls: [],
+    }
```

Comment:
This way we catch the error from AI apis providers
<img width="559" alt="image" src="https://github.com/user-attachments/assets/cfb3afdd-d2cc-43bb-9e24-135806355b64">